### DISCUSSION TITLE
Default plot style

### DISCUSSION THREAD

**Discussion by daniel-caichac-DHI** [DHI/modelskill#135] [Ideas]
Hi, is it ok if I change the Default style of the Scatter plots from  `2D histogram` to `Scatter Density`, and default cmap from `viridis` to `rainbow` ? I think that it would make quite a change, as of now the default is a 2D Histogram with viridis, see example:

![image](https://user-images.githubusercontent.com/97288080/191250545-811fe613-062f-4898-9023-1bbf982c0589.png)

* Whereas if we put scatter density and rainbow as default same plot looks like this 

![image](https://user-images.githubusercontent.com/97288080/191250880-e755aa36-aa51-4ec4-b377-0b4c7aa526b9.png)

* Maybe the viridis could stay as default (see next plot with viridis and scatter density) but the 2D histogram is not ideal as default I would say:

![image](https://user-images.githubusercontent.com/97288080/191250772-a021984c-47fa-482b-a297-1b5435cdfcaa.png)

* Opinions? I can make the change quickly

**Comment by jsmariegaard** [comment]
I wanted to change the default colormap earlier (and still do) - I got stuck in making it consistent with the plotly backend. See https://github.com/DHI/fmskill/pull/91 . There can be many opinions on the best cmap - personally, I like YlOrRd more, but we can cetainly agree that the current default is not good. @ecomodeller what do you say?

**Reply by ecomodeller** [reply]
We use a color scale in the scatter plot to indicate the relative number of points. Some areas have a lot of data, some less data and most have no data. So I think a very simple sequential color scale like `Blues` could be appropriate. 

More complicated colormaps like rainbow seems like a bad idea ([Don’t #1: Use the “rainbow” scale](https://blog.bioturing.com/2018/09/24/heatmap-color-scale#98af))

**Reply by daniel-caichac-DHI** [reply]
I agree with the colormap, I have seen these post before not with rainbow but with 'jet' colormap which is dreadful, so we can agree with the blues or so. Should we go for it? also what about the the density over 2d-Histogram?

**Reply by jsmariegaard** [reply]
I prefer something which ends in dark red as it quite clearly means "high intensity". In order cases, you would use Blue to indicate negative or low values... The reason why I previously used YlOrRd is because it fades naturally towards white (=no values) in the 2d histogram.

**Reply by jsmariegaard** [reply]
And please don't forget plotly - see #91 ....

**Reply by ecomodeller** [reply]
Transparent - Yellow - Red  (no need to state that Orange is between Yellow and Red😉) seems pretty common for heatmaps Example: https://www.kaggle.com/code/mateusaraujo0/uber-heatmap-deck-gl

**Reply by daniel-caichac-DHI** [reply]
What do you think about `plasma` as default cmap?  I have been using it already and it is one of the perceptually uniform colormaps from matplotlib
![image](https://user-images.githubusercontent.com/97288080/194526127-3059477a-2f40-4dbf-936b-09bc0e02741f.png)

I think that the high-density area should be something like bright yellow and the low density should be darker (as the camp I propose) and not the other way around as the uber cmap, otherwise, the points that are far away (~outliers) won't be noticed when in reality those could actually dominate design conditions or whatnot, so we **want** to see them; I don't want to have these points as transparent just because they have low density: eg, the 100yr Hm0 maybe be located on top of the 1:1 line (perfect model fit), but will have very low density (maybe just once in my simulation), hence I need to see this point, making it transparent is risky.

I think the uber heatmap is good for a spatial-heatmap, but not ideal for a scatter plot heatmap for our modelling-purposes

**Reply by jsmariegaard** [reply]
I think we may need different colormaps depending on whether it's a 2d-histogram-type (where transparent-yellow-red works well) and density-type where outliers also need to be highlighted (as you point out). I guess it also depends on whether you're interested in extremes or normal conditions.

**Reply by jsmariegaard** [reply]
But I think that `plasma` looks good for the density plot!

**Reply by jsmariegaard** [reply]
I think we should do something about the quantiles - they are too dominating this way. Maybe a piecewise linear line would be better?

**Reply by daniel-caichac-DHI** [reply]
> But I think that `plasma` looks good for the density plot!

Viridis actually also look quite decent so any of those is good with me

> I think we should do something about the quantiles - they are too dominating this way. Maybe a piecewise linear line would be better?

I agree with the QQ being too dominant, but I am not sure I follow with the piecewise linear (as in, a line instead of points?)



> I think we may need different colormaps depending on whether it's a 2d-histogram-type (where transparent-yellow-red works well) and density-type where outliers also need to be highlighted (as you point out). I guess it also depends on whether you're interested in extremes or normal conditions.

Agree
